### DISCUSSION TITLE
Some Notice

### DISCUSSION THREAD

**Discussion by shugen002** [cawa-93/vite-electron-builder#482] [Show and tell]
If you need to have more than one web-based package and require to add to watch, please remember to set a different `cacheDir` in `vite.config.js` , or they will override each other.

```js
// vite.config.js in renderer
/* eslint-env node */

import { chrome } from '../../electron-vendors.config.json';
import { builtinModules } from 'module';
import vue from '@vitejs/plugin-vue';

const PACKAGE_ROOT = __dirname;

/**
 * @type {import('vite').UserConfig}
 * @see https://vitejs.dev/config/
 */
const config = {
  mode: process.env.MODE,
  root: PACKAGE_ROOT,
  plugins: [vue()],
  base: '',
  server: {
    fs: {
      strict: true,
    },
  },
  build: {
    sourcemap: true,
    target: `chrome${chrome}`,
    outDir: 'dist',
    assetsDir: '.',
    terserOptions: {
      ecma: 2020,
      compress: {
        passes: 2,
      },
      safari10: false,
    },
    rollupOptions: {
      external: [
        ...builtinModules,
      ],
    },
    emptyOutDir: true,
    brotliSize: false,
  },
  cacheDir: '../../node_modules/.vite/renderer',
};

export default config;
``` 

```js
// vite.config.js in web
/* eslint-env node */

import { chrome } from '../../electron-vendors.config.json';
import { join } from 'path';
import { builtinModules } from 'module';
import vue from '@vitejs/plugin-vue';

const PACKAGE_ROOT = __dirname;

/**
 * @type {import('vite').UserConfig}
 * @see https://vitejs.dev/config/
 */
const config = {
  mode: process.env.MODE,
  root: PACKAGE_ROOT,
  resolve: {
    alias: {
      '/@/': join(PACKAGE_ROOT, 'src') + '/',
    },
  },
  plugins: [vue()],
  base: '',
  server: {
    fs: {
      strict: true,
    },
  },
  build: {
    sourcemap: true,
    target: `chrome${chrome}`,
    outDir: 'dist',
    assetsDir: '.',
    terserOptions: {
      ecma: 2020,
      compress: {
        passes: 2,
      },
      safari10: false,
    },
    rollupOptions: {
      external: [
        ...builtinModules,
      ],
    },
    emptyOutDir: true,
    brotliSize: false,
  },
  cacheDir: '../../node_modules/.vite/web',
};

export default config;
```

by the way , should we add a example include a simple server and page?

**Comment by shugen002** [comment]
@cawa-93 after some development , i think whether we add Package 'web' or not , we still need to set `cacheDir`. Bacause no matter which  package, `main` , `preload` or `renderer` building , they will override other's cache.

**Reply by cawa-93** [reply]
An interesting remark. How can I test whether the cache of each package overwrites each other?

**Reply by shugen002** [reply]
sorry about this miss idea. after I did more investgate about that and notice the `main` and `preload` didn't use cacheDir at all.

**Reply by shugen002** [reply]
I will delete this reply 24hours later.

**Comment by shugen002** [comment]
One more thing, i have no idea is this an expect behavior or my fail. when i use some node modules in `main` , I need to manual add the module name into `packages\main\vite.config.js` `config.build.rollupOptions.external` . Maybe we can read dependence from  `package.json` ?

**Reply by cawa-93** [reply]
I prefer the opposite: to pack all the dependencies into bundle. This improves by tree-shaking. But, to my great regret, there are packages that [can not be bundled](https://github.com/npm/node-semver/issues/381) (due to the specific architecture, etc.). Therefore, such packages are **forced** to be delivered as external.

**Reply by shugen002** [reply]
I agree that a more elegant solution , but we still need to face current situation.

According to current electron-builder behavior , it will copy `dependencies` in package.json into final product. It mean that we can maintain this **forced** packages while telling electron-builder copy these packages.

By the way, maybe you should move `vue` and `vue-router` into `devDependencies`.

**Comment by udondan** [comment]
But how would you know which packages are meant for main?

**Reply by shugen002** [reply]
I keep only packages required by `main` in `dependencies`, and the rest in devDependecies for smaller package size (mainly effect `app.asar`).

By the way , i am looking for a easy way to compile Electron without some ability to shrink the size of electron.

**Reply by cawa-93** [reply]
This could be solved by completely dividing the project into packages and using npm workspaces. But the last time I tried it caused [problems](https://github.com/electron-userland/electron-builder/issues/5713)
